### PR TITLE
RemoteRendering: Fix for pylint error: Use CaseInsensitiveEnumMeta

### PR DESCRIPTION
**Pull request by ChristopherManthei**
# Description

Fixes pylint error: Use CaseInsensitiveEnumMeta.
This allows customers to ignore casing when setting enum values.

### CODE REVIEW

**Review by MichaelZp0 — APPROVED** [review]

File: sdk/remoterendering/azure-mixedreality-remoterendering/CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
 # Release History
 
-## 1.0.0b3 (Unreleased)
-
-## 1.0.0b2 (Unreleased)
+## 1.0.0b2 (2023-06-12)
```
Comment (tjprescott) [inline sdk/remoterendering/azure-mixedreality-remoterendering/CHANGELOG.md:3]: When do you actually plan to release this?

**Review by MalcolmTyrrell — APPROVED** [review]

File: sdk/remoterendering/azure-mixedreality-remoterendering/CHANGELOG.md (hunk 1 above)
Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/CHANGELOG.md]: Changed release date to Monday. Will create an out-of-band release then.

**Review by FlorianBorn71 — APPROVED** [review]

File: sdk/remoterendering/azure-mixedreality-remoterendering/pyproject.toml [hunk 2]
```diff
@@ -2,3 +2,4 @@
 pyright = false
 type_check_samples = false
 verifytypes = false
+pylint = false
```
Comment (kashifkhan) [inline sdk/remoterendering/azure-mixedreality-remoterendering/pyproject.toml:5]: Any particular reason to disable pylint ?

Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/pyproject.toml:5]: I followed the steps that were outlined in the doc linked by the pylint error. The result: pylint crashes now on the repo. It looks like we are hitting [pylint-dev/pylint#5935](https://github.com/pylint-dev/pylint/issues/5935)
```
"TypeError: argument of type 'Const' is not iterable"
```
As we can't upgrade pylint the only way to fix this is to disable it entirely from what I can tell. Any suggestions how to fix this?

File: sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py [hunk 3]
```diff
@@ -5,9 +5,11 @@
 # --------------------------------------------------------------------------
 
 from enum import Enum
+from six import with_metaclass
```
[the remark this answers is not in the document]
Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py:8]: This is what the pylint error told me to do: [[C4747(enum-must-inherit-case-insensitive-enum-meta), RemoteRenderingApiVersion] The enum must inherit from CaseInsensitiveEnumMeta](https://azure.github.io/azure-sdk/python_implementation.html#extensible-enumerations). Can you link me a guide how to do this differently?

File: sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py [hunk 4]
```diff
@@ -5,9 +5,11 @@
...
 
+from azure.core import CaseInsensitiveEnumMeta
 
-class RemoteRenderingApiVersion(str, Enum):
+class RemoteRenderingApiVersion(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
```
[the remark this answers is not in the document]
Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py:12]: Oh I see, I'll give that a try, I was merely following the [[C4747(enum-must-inherit-case-insensitive-enum-meta), RemoteRenderingApiVersion] The enum must inherit from CaseInsensitiveEnumMeta](https://azure.github.io/azure-sdk/python_implementation.html#extensible-enumerations) page linked in the pylint error.

File: sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py]: Thanks! I've made your suggested changes and re-enabled pylint, maybe it doesn't crash on the new syntax.

File: sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (kashifkhan) [reply sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py]: @ChristopherManthei Thanks for brining that up, it needs to be updated from our end :) 

Your changes will be exactly like the Enums here https://github.com/Azure/azure-sdk-for-python/blob/e7da08142120fb4a1466f408acf38d8c2d584524/sdk/servicebus/azure-servicebus/azure/servicebus/_common/constants.py#L148

File: sdk/remoterendering/azure-mixedreality-remoterendering/pyproject.toml (hunk 2 above)
Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/pyproject.toml]: Seems to work again now that the code has changed. So pylint is enabled again.

File: sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/azure/mixedreality/remoterendering/_api_version.py]: Fixed!

**Pull request merged**